### ISSUE TITLE
FP become slower after upgrade to 0.4

### ISSUE THREAD

**Issue by panjunwen** [gpleiss/efficient_densenet_pytorch#38]
Hi,
Thanks for your works!
Recently I upgrade my network to 0.4 with your implementation of DenseNet. And I found that the new version is slower than before. I thought that the shared memory could speed up the forward pass obviously.In my application, predicting one subject on the 0.3.x version cost 9s but now it need 11s.

The dice metric also get worth than before. I found that in the new code you use the Kaiming normal initialization but before default initialization (uniform?). I have try to make all parameters as before but it has not effect. Have you some advice for me?

Thanks.

**Comment by gpleiss** [comment]
Were you using the efficient implementation before? I think efficient is true by default.
The speed of the efficient version now depends on how much memory you have available. The more memory, the faster it runs.

What is the "dice" metric?

**Comment by panjunwen** [comment]
Thanks for your reply.
Yes. I use your efficient implementation before. 
I found that the forward pass cost more time now, and I think the shared memory may speed up the forward pass.

**Comment by panjunwen** [comment]
Dice == F1-score.
I use it to evaluate my network. I'm confused that the performance get worse after I update to 0.4.x.

**Comment by panjunwen** [comment]
I apply DenseNet on 3D medical image segmentation.
F1-score is computed in numpy on per-class. Data set is labeled by myself.
Everything works well with pytorch 0.3.1 and your efficient implementation before. I re-read the paper and found that you use MSRA inialization and now I can confirm that initialization is not the problem. 
I check the PyTorch code about checkpointing. It just drop the intermediate results and re-compute them when back-prop. That's a really simple method and there's no shared memory. I  care the test speed instead of training. I think that the shared memory is useful to speed up the forward pass. 
I have transfer the parameters from 0.3.1 version to lastest and the network works bad. All parameters are same as before but the result is different. I will try to locate the problem.

Thanks.

**Comment by panjunwen** [comment]
I locate the main problem and everything works well now.
I'm misleaded by this tutorial [https://pytorch.org/2018/04/22/0_4_0-migration-guide.html].
Thanks for your reply.

**Comment by panjunwen** [comment]
I locate the main problem and everything works well now.
I'm misleaded by this tutorial [](https://pytorch.org/2018/04/22/0_4_0-migration-guide.html).
Thanks for your reply.
